### PR TITLE
ROE-150 Separate the DTO and DAO models

### PR DESCRIPTION
**Pull request by mwestacott**
* Implementation in line with recommendation from Sonar
* DTO (JSON) model classes now added
* DAO (Mongo) model classes also added
* Mapper called from service to move JSON data into Mongo object/format
* Unit test added

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -17,6 +17,7 @@
         <maven-surefire-plugin.version>2.22.2</maven-surefire-plugin.version>
         <structured-logging.version>1.9.14</structured-logging.version>
         <log4j.version>2.17.2</log4j.version>
+        <org.mapstruct.version>1.5.0.RC1</org.mapstruct.version>
```
Comment (mwestacott) [inline pom.xml:20]: We used version 1.4.2 on Confirmation Statement but used the latest version available here.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/EntityDto.java [hunk 2]
```diff
@@ -1,63 +1,41 @@
...
+    @JsonProperty("legal_form")
     private String legalForm;
 
-    @JsonProperty(LAW_GOVERNED_FIELD)
-    @Field(LAW_GOVERNED_FIELD)
+    @JsonProperty("law_governed")
     private String lawGoverned;
 
-    @JsonProperty(PUBLIC_REGISTER_ENTITY_REGISTERED_ON_FIELD)
+    @JsonProperty("public_register_entity_registered_on")
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/EntityDto.java:34]: public_register_entity_registered_on is not meant to be a date, it is just known as 'publicRegister' in the web app - 'On which public register is the overseas entity registered'
probably needs renaming

Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/EntityDto.java:34]: public_register_entity_registered_on is not meant to be a date, it is just known as 'publicRegister' in the web app - 'On which public register is the overseas entity registered'
probably needs renaming

Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/EntityDto.java:34]: Duplicate comment? Will resolve this convo.

Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/EntityDto.java]: Fixed. Updated the spec with the changed field name too.

**Review by markpit — APPROVED** [review]

**Pull request merged**